[WAZO-3815] Introduce docusaurus beta docs

Romej opened the pull request:
## Summary of changes
- introduce a `website` folder which contains new docusaurus beta docs
- graphs migrated to mermaid
- minor tweaks to make docs more cohesive
- migrated elements : 
    - [X] homepage / intro
    - [X] API overviews / dev notes / admin notes / glossary
    - [X] blog
    - [X] release notes
- elements that will be migrated in the next phase:
    - uc-doc

## How to test
- `cd website && yarn && yarn start`
- existing gatsby site should not be affected
- please don't flag broken links, they are expected at this stage

> [!NOTE]  
> Beta docs are intended to be made available to everyone through an upcoming `beta.wazo-platform.org` subdomain. The current docs should remain accessible until all of the content has been migrated

chartrandf reviewed (commented):
Ohh yes, thanks for starting this transition 🙏 

Adjustments to consider:
- Should we [versioning](https://docusaurus.io/docs/versioning) documentation?
  - If yes, we should also check for [multi-instance](https://docusaurus.io/docs/docs-multi-instance). Example no version for blog, but we version the rest of the doc.
- `/blog` easy to have a summary? The listing is very long
- Homepage is missing some content, normal?

Romej commented:
>     * Should we [versioning](https://docusaurus.io/docs/versioning) documentation?

docusaurus warns that versioning introduces complexity. I do however agree that it's relevant to our situation, but i think it should be done one we have a fully migrated version that works. I added it to the sub-tasks : https://wazo-dev.atlassian.net/browse/WAZO-3818

>     * `/blog` easy to have a summary? The listing is very long
Not very easy as we need to use [`<!-- truncate -->`](https://docusaurus.io/docs/blog#blog-list)) individually in each blog post to determine what part should be part of the summary.

>     * Homepage is missing some content, normal?
I moved some of the homepage content to `/docs/intro`

chartrandf commented:
> > ```
> > * `/blog` easy to have a summary? The listing is very long
> > ```
> 
> 
> Not very easy as we need to use [`<!-- truncate -->`](https://docusaurus.io/docs/blog#blog-list)) individually in each blog post to determine where the summary ends. Maybe it's a task that we can split among contributors

Okay, added `truncate` to first 10 articles. Let's improve this in the futur :)
